How-to on creating a did web (#620)
TBD54566975/ssi-service: doc/howto/didweb.md

# How To: Create a did:web

## Background

> A [did:web Method Specification)](https://w3c-ccg.github.io/did-method-web/) describes a DID method that uses the web domain's existing reputation.

Reviewer: ```suggestion
The [did:web Method Specification)](https://w3c-ccg.github.io/did-method-web/) describes a DID method that uses an existing web domain to host and establish trust for a DID Document.
```

> It relies on the controller of an existing domain to host a custom file with the contents of the DID Document they want to expose. SSI Services facilitates creation of a `did:web`.

Reviewer: ```suggestion
It relies on the controller of an existing domain to host a custom file with the contents of the DID Document they want to expose. The SSI Service facilitates creation of a `did:web`, which you then must update on the domain you control.
```

## Steps

### Prerequisites

* You control an existing domain  (e.g. like https://www.tbd.website).

> * You are able to host files in a path within that origin.(e.g. you can host the file returned by https://www.tbd.website/.well-known/did.json)

Reviewer: ```suggestion
* You are able to host files in a path within that origin (e.g. you can host the file returned by https://www.tbd.website/.well-known/did.json).
```

## 1. Create a `did:web` DID

> Make a `PUT` request to `/v1/dids/web`:

Reviewer: ```suggestion
Make a `PUT` request to `/v1/dids/web`, with a request body as follows:
```

```json
{

>   "keyType":"Ed25519",

Reviewer: ```suggestion
  "keyType": "Ed25519",
```

  "options": {
    "didWebId": "did:web:tbd.website"
  }
}
```

Or if you like CURLing:

```shell
curl -X PUT 'localhost:3000/v1/dids/web' -d '{
  "keyType":"Ed25519",
  "options": {
    "didWebId": "did:web:tbd.website"
  }
}'
```

> If successful, the contents of the response should look like the contents below...

Reviewer: ```suggestion
Upon success, the contents of the response should look as follows...
```

```json
{
  "did": {
    "@context": "https://www.w3.org/ns/did/v1",
    "id": "did:web:tbd.website",
    "verificationMethod": [
      {
        "id": "did:web:tbd.website",
        "type": "JsonWebKey2020",
        "controller": "did:web:tbd.website#owner",
        "publicKeyJwk": {
          "kty": "OKP",
          "crv": "Ed25519",
          "x": "TuAM4Ro4q5_cFMarCHmOm-1c7NaxBxvoEe7-x7K7xhw",
          "alg": "EdDSA",
          "kid": "did:web:tbd.website#owner"
        }
      }
    ],
    "authentication": [
      [
        "did:web:tbd.website#owner"
      ]
    ],
    "assertionMethod": [
      [
        "did:web:tbd.website#owner"
      ]
    ]
  }
}
```

> This response is a and object containing a `did` property, whose value is a DID Document. Such value is what needs to be hosted.

Reviewer: ```suggestion
This response is an object containing a `did` property, whose value is a DID Document. This value is what needs to be hosted on your domain.
```
